operator/internal/controller/pulumi/session_test.go

```diff
@@ -357,17 +359,17 @@ func TestSetupWorkspace(t *testing.T) {
 						},
 					},
 				},
-				WorkspaceTemplate: &autov1alpha1.EmbeddedWorkspaceTemplateSpec{
-					Metadata: autov1alpha1.EmbeddedObjectMeta{
+				WorkspaceTemplate: &shared.WorkspaceApplyConfiguration{
+					ObjectMetaApplyConfiguration: &metav1apply.ObjectMetaApplyConfiguration{
```

Comment:
Worth calling out that `ObjectMetaApplyConfiguration` differs from `EmbeddedWorkspaceTemplateSpec` in that it does expose all of the underlying fields (like generation) even if they can't actually be set. I think it makes for a more bloated/confusing schema but I don't see a good way around it.

Reply:
This approach is similar to `PodTemplateSpec` so it shouldn't add confusion to K8s users.